### PR TITLE
bugfix: Add validation for IntorString in health check

### PR DESCRIPTION
**Pull request by shitaljante**
### What type of PR is this?
_(bug)_

### What this PR does / why we need it?
Add validation for IntorString type in MaxUnavailable specs

### Which Jira/Github issue(s) this PR fixes?
[OSD-24726](https://issues.redhat.com//browse/OSD-24726)

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/upgraders/healthcheck_pdb.go [hunk 1]
```diff
@@ -44,11 +45,31 @@ func checkPodDisruptionBudgets(c client.Client, logger logr.Logger) (string, err
...
+				}
+			}
+			if pdb.Spec.MaxUnavailable != nil && value == "0" {
+				//misconfigured pdb
+				fmt.Printf("PodDisruptionBudget: %s/%s\n", pdb.Namespace, pdb.Name)
+				return metrics.ClusterInvalidPDBConf, fmt.Errorf("found a PodDisruptionBudget with MaxUnavailable set to 0")
+			}
+		}
+
+		for _, pdb := range pdbList.Items {
```
Comment (tkong-redhat) [inline pkg/upgraders/healthcheck_pdb.go:65]: Do we need a second loop here?

File: pkg/upgraders/healthcheck_pdb.go [hunk 2]
```diff
@@ -44,11 +45,31 @@ func checkPodDisruptionBudgets(c client.Client, logger logr.Logger) (string, err
 		logger.Info("unable to list PodDisruptionBudgets/v1")
 		return metrics.PDBQueryFailed, err
 	}
-
+	var value string
 	for _, pdb := range pdbList.Items {
 		if !strings.HasPrefix(pdb.Namespace, "openshift-*") || checkNamespaceExistsInArray(namespaceException, pdb.Namespace) {
-			if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntVal == 0 {
+			if pdb.Spec.MaxUnavailable != nil {
```
Comment (tkong-redhat) [inline pkg/upgraders/healthcheck_pdb.go:51]: I'm OK with the current implementation. 
Just wonder if this is better
`if pdb.Spec.MaxUnavailable != nil && ((pdb.Spec.MaxUnavailable.Type == intstr.Int) && (pdb.Spec.MaxUnavailable.IntVal == 0)) `

Comment (shitaljante) [reply pkg/upgraders/healthcheck_pdb.go:51]: yeah this can be implemented. For safe side I added for string "0" as well.

Comment (tkong-redhat) [reply pkg/upgraders/healthcheck_pdb.go]: Just a discussion. Thanks for taking it. 
Your concern is valid. However, I remember user is not able to set string "0" here.

**Comment by tkong-redhat** [comment]
\lgtm

**Comment by tkong-redhat** [comment]
\approve

**Comment by tkong-redhat** [comment]
\approved

File: pkg/upgraders/healthcheck_pdb.go [hunk 3]
```diff
@@ -44,20 +45,24 @@ func checkPodDisruptionBudgets(c client.Client, logger logr.Logger) (string, err
...
 	}
-
 	for _, pdb := range pdbList.Items {
 		if !strings.HasPrefix(pdb.Namespace, "openshift-*") || checkNamespaceExistsInArray(namespaceException, pdb.Namespace) {
-			if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntVal == 0 {
-				//misconfigured pdb
+			if pdb.Spec.MaxUnavailable != nil && ((pdb.Spec.MaxUnavailable.Type == intstr.Int) && (pdb.Spec.MaxUnavailable.IntVal == 0)) {
+				logger.Info(fmt.Sprintf("PodDisruptionBudget: %s/%s\n", pdb.Namespace, pdb.Name))
 				return metrics.ClusterInvalidPDBConf, fmt.Errorf("found a PodDisruptionBudget with MaxUnavailable set to 0")
+			} else if pdb.Spec.MinAvailable != nil && ((pdb.Spec.MinAvailable.Type == intstr.String) && (pdb.Spec.MinAvailable.StrVal == "100%")) {
```
Comment (tkong-redhat) [inline pkg/upgraders/healthcheck_pdb.go:53]: Shall we check the Min and Max both? 
In this logic, MinAvailable only be checked when MaxAvailable check success.
What do you think?

Comment (shitaljante) [reply pkg/upgraders/healthcheck_pdb.go]: good point, updated changes

**Comment by tkong-redhat** [comment]
\lgtm

**Pull request merged**